### DISCUSSION TITLE
Should PowerShell join the .NET foundation or create its own?

### DISCUSSION THREAD

**Discussion by kilasuit** [PowerShell/PowerShell#20773] [General]
Bear in mind there is already PowerShell.org which could be seen as filling this role already.

I think as a .NET language joining the .NET Foundation kinda makes sense and that the role of PowerShell.org should stay as is alongside additional other regional similar entities going forward that will help drive regional growth of community events and development going forward into future years.

**Comment by fMichaleczek** [comment]
This is just an opinion, which surely lacks discernment, but if I had the power to decide, I would do this:

- separate PowerShell from System.Management.Automation
- Produce a monadium with SMA (no CIM, DSC, COM+, no modules, no PowerShell Gallery).

At that point, Microsoft would have more freedom to be conservative with its PowerShell product (it can choose the Monadium branch) and can let the core develop without needing an army to validate a patch.

This project to make SMA modular can take 1 year of work from an experienced person to reassemble an SDK and migrate SMA code that should not be there.

The benefit of joining the dotnet foundation is access to Arcade. There may be no interest for the PowerShell team but for a Monadium surely. If the goal is to be closer to dotnet than PowerShell and to implement features from dotnet instead of focusing on modules.
https://github.com/dotnet/arcade/tree/main

Why, would you tell me? For the same reason that Rust doesn't have builtin libraries and dotnet provides a framework.

The best solution is to stop providing modules and focus on the dotnet bridge.
